src/lib/replication/repl_dev/common.cpp

```diff
@@ -91,6 +100,19 @@ ReplServiceError repl_req_ctx::alloc_local_blks(cshared< ReplDevListener >& list
     auto const hints_result = listener->get_blk_alloc_hints(m_header, data_size);
     if (hints_result.hasError()) { return hints_result.error(); }
 
+    if (hints_result.value().committed_blk_id.has_value()) {
+        //if the allocted_blk_id is already present, use it and skip allocation
+        LOGINFO("For Repl_key=[{}] data already exists, skip", rkey().to_string());
+        m_local_blkid = hints_result.value().committed_blk_id.value();
+        add_state(repl_req_state_t::BLK_ALLOCATED);
+        add_state(repl_req_state_t::DATA_RECEIVED);
+        add_state(repl_req_state_t::DATA_WRITTEN);
+        add_state(repl_req_state_t::DATA_COMMITTED);
+        m_data_received_promise.setValue();
+        m_data_written_promise.setValue();
+        return ReplServiceError::OK;
```

Comment:
if we return here, we will not set m_local_blkid in line 117, it is very critical for log entry.

for every member(leader or follower), if a log entry is data_linked, then the log entry will have a member of local_blk_id to indicate the data associated with this log entry.

when leader send log entry to follower, the local_blk_id will be considered as remote_blk_id in the follower side , used for fetch data.  when follower append log to local log store, it will change the local_blk_id to its own local_blk_id, so that when this follower becomes leader, it can send this log to other follower , and other follower can fetch data from this new leader according to the local_blk_id in this log entry.

so here, we need to change m_local_blkid to the commited_blk_id

Reply:
Yes, I have already set `m_local_blkid` to the `commited_blk_id` at line 106.